### PR TITLE
Minor Fable AST cleanup

### PR DESCRIPTION
**Pull request by ncave**
- Just so it doesn't get lost in the sands of time, a modest proposal to make `Fable.Set` a bit more symmetric with `Fable.Get`.

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
We can create a new `next` branch for next major release and merge this there. Other changes we may want to do to the AST:

- Give names to all the union fields. This way plugins can pattern match union fields by name instead of position so hopefully they don't break if we add a new field to the union case.
- Add union case tag to `Get(_, UnionField(index, t), _, _)`, as in the F# AST, so we can get the union case and field from the entity reference. See https://github.com/fable-compiler/Fable/pull/2447#discussion_r638436247
- Add metadata to all elements so we can "extend" the AST (at the cost of some type safety) without modifying it.

**Comment by ncave** [comment]
@alfonsogarciacaro Perhaps also add entity reference to `AnonymousRecordType`:
```fs
| AnonymousRecordType of EntityRef * fieldNames: string [] * genericArgs: Type list
```

File: src/Fable.AST/Fable.fs [hunk 1]
```diff
@@ -266,12 +267,18 @@ type KeyKind =
 type GetKind =
     | ByKey of KeyKind
     | TupleIndex of int
+    | FieldGet of Field * index: int
```
Comment (alfonsogarciacaro) [inline src/Fable.AST/Fable.fs:270]: Should we add `Field` to the AST elements or maybe just the name so we can retrieve the field from the entity reference when needed as in here? https://github.com/fable-compiler/Fable/pull/2447#discussion_r640307157

Comment (alfonsogarciacaro) [reply src/Fable.AST/Fable.fs:270]: Same for `SetKind` and `KeyKind` (which already contain a direct reference to `Field` but maybe we can remove it).

**Review by alfonsogarciacaro — COMMENTED** [review]
Sorry @ncave I created the `next` branch and merged this too quickly but I've left a comment later. Could you please check? Thanks!

File: src/Fable.AST/Fable.fs (hunk 1 above)
Comment (ncave) [reply src/Fable.AST/Fable.fs:270]: @alfonsogarciacaro I think adding Field is fine, but whatever yo prefer.

**Pull request merged**